public/app/plugins/datasource/loki/components/useLokiSyntax.ts

```diff
@@ -0,0 +1,55 @@
+import { useState, useEffect } from 'react';
+import LokiLanguageProvider from 'app/plugins/datasource/loki/language_provider';
+import Prism from 'prismjs';
+import { useLokiLabels } from 'app/plugins/datasource/loki/components/useLokiLabels';
+import { CascaderOption } from 'app/plugins/datasource/loki/components/LokiQueryFieldForm';
+
+const PRISM_SYNTAX = 'promql';
+
+/**
+ *
+ * @param languageProvider
+ * @description Initializes given language provider, exposes Loki syntax and enables loading label option values
+ */
+export const useLokiSyntax = (languageProvider: LokiLanguageProvider) => {
+  // State
+  const [languageProviderInitialized, setLanguageProviderInitilized] = useState(false);
+  const [syntax, setSyntax] = useState(null);
+
+  /**
+   * Holds information about currently selected option from rc-cascader to perform effect
+   * that loads option values not fetched yet. Based on that useLokiLabels hook decides whether or not
+   * the option requires additional data fetching
+   */
+  const [activeOption, setActiveOption] = useState<CascaderOption[]>();
+
+  const { logLabelOptions, setLogLabelOptions, refreshLabels } = useLokiLabels(
+    languageProvider,
+    languageProviderInitialized,
+    activeOption
+  );
+
+  // Async
+  const initializeLanguageProvider = async () => {
+    if (!languageProviderInitialized) {
```

Comment:
It seems like this effect will be only fired once and `languageProviderInitialized` is only set here so maybe this check is unnecessary?

Reply:
👍